src/main/java/com/onaio/steps/handler/actions/SubmitDataHandler.java

```diff
@@ -173,7 +174,13 @@ public void enable() {
     }
 
     public void displayUploadResult(List<UploadResult> uploadResults) {
-        HouseholdUploadResultDialog householdUploadResultDialog = new HouseholdUploadResultDialog(uploadResults);
-        householdUploadResultDialog.show(activity.getSupportFragmentManager(), HouseholdUploadResultDialog.class.getSimpleName());
+        try {
+            HouseholdUploadResultDialog householdUploadResultDialog = new HouseholdUploadResultDialog(uploadResults);
+            FragmentTransaction fragmentTransaction = activity.getSupportFragmentManager().beginTransaction();
+            fragmentTransaction.add(householdUploadResultDialog, HouseholdUploadResultDialog.class.getSimpleName());
+            fragmentTransaction.commitAllowingStateLoss();
+        } catch (Exception ex) {
```

Comment:
Remove the unspecified `try { } catch (Exception)`.

Why is this error happening?

Reply:
previously, having a problem with the `commit()` method if we destroy the activity before showing the dialog fragment. The issue was fixed when i replace the `commit` method with `commitAllowingStateLoss`. Code updated with and try-catch now removed

Reply:
OK, from my understanding. This dialog is only shown once the upload is done. You are saying that you get a crash when the activity finishes before showing the dialog fragment, I don't understand if this is right.

On the other hand, I believe this error occurs if you try to show the dialog when the activity is either paused or finishing. Is there a better way to prevent showing the dialog if the activity is in either states?